### PR TITLE
PSVAMB-42565 -> Change Cortex provider to work with files that stored…

### PR DESCRIPTION
**Pull request by TalDubovKaltura**
… in AWS S3

### CODE REVIEW

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php [hunk 1]
```diff
@@ -7,6 +7,7 @@ class CortexApiDistributionEngine extends DistributionEngine implements
 	IDistributionEngineSubmit,
 	IDistributionEngineDelete
 {
+    protected $tempXmlPath;
```
Comment (inbal-ben-david) [inline plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php:10]: indentation

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php [hunk 2]
```diff
@@ -34,6 +35,21 @@ class CortexApiDistributionEngine extends DistributionEngine implements
 	 * @var string
 	 */
 	private $recordId;
+
+    public function configure()
```
Comment (inbal-ben-david) [inline plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php:39]: indentation

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php [hunk 3]
```diff
@@ -407,8 +423,42 @@ protected function uploadVideo(KalturaCortexApiDistributionJobProviderData $apiD
 	{
 		try
 		{
-			$SystemIdentifier = $this->mediaUpload($apiDistributionJobProviderData->videoAssetFilePath);
+            $needDel = false;
+            $videoFilePath = $apiDistributionJobProviderData->videoAssetFilePath;
+            if (!$videoFilePath)
```
Comment (inbal-ben-david) [inline plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php:428]: add {} for all if statements here

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php [hunk 4]
```diff
@@ -407,8 +423,42 @@ protected function uploadVideo(KalturaCortexApiDistributionJobProviderData $apiD
...
+
+            if (!kFile::checkFileExists($videoFilePath))
+                throw new KalturaDistributionException('The file ['.$videoFilePath.'] was not found (probably not synced yet), the job will retry');
+
+            list($isRemote, $remoteUrl) = kFile::resolveFilePath($videoFilePath);
+            $tempVideoFilePath = !$isRemote ? null : kFile::getExternalFile($remoteUrl, null, basename($videoFilePath));
+
+            if (FALSE === strstr($videoFilePath, "."))
+            {
+                $videoFilePathNew = $this->tempXmlPath . "/" . uniqid() . ".dme";
```
Comment (inbal-ben-david) [inline plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php:439]: i dont think that the dme makes sense here, its is part of the path for daily motion distribution but how it connects here?

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php [hunk 5]
```diff
@@ -407,8 +423,42 @@ protected function uploadVideo(KalturaCortexApiDistributionJobProviderData $apiD
...
+                {
+                    $copyFrom = $tempVideoFilePath ? $tempVideoFilePath : $videoFilePath;
+                    kFile::copy($copyFrom,$videoFilePathNew);
+                    $needDel = true;
+                }
+                $videoFilePath = $videoFilePathNew;
+            }
+            elseif($isRemote)
+            {
+                $videoFilePath = $tempVideoFilePath;
```
Comment (inbal-ben-david) [inline plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php:450]: indentation

File: plugins/event_notification/providers/http/lib/KDispatchHttpNotificationEngine.class.php [hunk 6]
```diff
@@ -88,6 +88,8 @@ public function sendHttpRequest(KalturaHttpNotificationTemplate $httpNotificatio
 		}
 
 		$curlWrapper = new KCurlWrapper();
+       		$curlWrapper->setOpt(CURLOPT_RETURNTRANSFER, 1);
```
Comment (inbal-ben-david) [inline plugins/event_notification/providers/http/lib/KDispatchHttpNotificationEngine.class.php:91]: why this change required for all http notifications?

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php (hunk 1 above)
Comment (TalDubovKaltura) [reply plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php:10]: Fixed here - https://github.com/kaltura/server/pull/12079/commits/d26def18891e9adca0d378d5a70af1d68110a07a

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php (hunk 3 above)
Comment (TalDubovKaltura) [reply plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php]: Fixed here - https://github.com/kaltura/server/pull/12079/commits/d26def18891e9adca0d378d5a70af1d68110a07a

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php (hunk 5 above)
Comment (TalDubovKaltura) [reply plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php]: Fixed here - https://github.com/kaltura/server/pull/12079/commits/d26def18891e9adca0d378d5a70af1d68110a07a

File: plugins/event_notification/providers/http/lib/KDispatchHttpNotificationEngine.class.php (hunk 6 above)
Comment (TalDubovKaltura) [reply plugins/event_notification/providers/http/lib/KDispatchHttpNotificationEngine.class.php]: Fixed. Sorry, this file change was relevant for another PR, I removed this change from here

File: plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php (hunk 4 above)
Comment (TalDubovKaltura) [reply plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php]: This logic was also in YoutubeAPI/DailyMotion, so Im not sure why is it necessary but I already showed this change to Yossi P. who approved this is the correct logic

**Review by inbal-ben-david — APPROVED** [review]

**Comment by TalDubovKaltura** [comment]
merged here: https://github.com/kaltura/server/pull/12084
